app/lib/frontend/handlers/headers.dart

```diff
@@ -13,21 +13,36 @@ class CacheHeaders {
   /// The maximum age while a response may be cached.
   final Duration maxAge;
 
-  /// The cache storage category used for signed-in users.
-  /// `private` by default.
-  final String? signedInStorage;
+  /// Add the `private` directive, indicating that the response must not be
+  /// stored in public cache (such as a CDN).
+  ///
+  /// If neither [private] or [public] is specified, `Cache-Control` will
+  /// set `private` if the request was authenticated.
+  final bool private;
+
+  /// Add the `public` directive, indicating that the response may be
+  /// stored in public cache (such as a CDN).
+  ///
+  /// If neither [private] or [public] is specified, `Cache-Control` will
+  /// set `private` if the request was authenticated.
+  final bool public;
 
   CacheHeaders._(
     this.maxAge, {
-    this.signedInStorage,
-  });
+    bool private = false,
+    bool public = true,
+  })  : public = public,
+        private = private {
+    assert(!private && !public, 'both private and public is not allowed');
+  }
 
   Map<String, String> call() {
     return <String, String>{
       HttpHeaders.cacheControlHeader: <String>[
-        requestContext.isNotAuthenticated
-            ? 'public'
-            : (signedInStorage ?? 'private'),
+        if (private || (!public && requestContext.isNotAuthenticated))
```

Comment:
I find this hard to parse (too many negations).

What happens if neither `private` nor `public` but `requestContext.isAuthenticated` - is that possible? Will it end up public?

Reply:
Maybe something along:
```suggestion
        if (public || requestContext.isNotAuthenticated)
          'public'
        else
          'private',
```

Reply:
If `private` then we force it 'private'
else if `public` then we make it 'public'
else if requestContext.isNotAuthenticated then we make it 'public'
else 'private'

Reply:
That's not what I read. I see the last `else` is `'public'`.